src/main/java/software/aws/neptune/gremlin/resultset/GremlinResultSetGetTypeInfo.java

```diff
@@ -0,0 +1,159 @@
+        INTEGER_INFO.put("PRECISION", 11);
+        INTEGER_INFO.put("LITERAL_PREFIX", null);
+        INTEGER_INFO.put("LITERAL_SUFFIX", null);
+        INTEGER_INFO.put("CASE_SENSITIVE", false);
+        INTEGER_INFO.put("MINIMUM_SCALE", 0);
+        INTEGER_INFO.put("MAXIMUM_SCALE", 0);
+        INTEGER_INFO.put("NUM_PREC_RADIX", 10);
+        TYPE_INFO.add(INTEGER_INFO);
+
+        SHORT_INFO.put("TYPE_NAME", "Short");
+        SHORT_INFO.put("DATA_TYPE", Types.SMALLINT);
+        SHORT_INFO.put("PRECISION", 5);
+        SHORT_INFO.put("LITERAL_PREFIX", null);
+        SHORT_INFO.put("LITERAL_SUFFIX", null);
+        SHORT_INFO.put("CASE_SENSITIVE", false);
+        INTEGER_INFO.put("MINIMUM_SCALE", 0);
+        INTEGER_INFO.put("MAXIMUM_SCALE", 0);
+        INTEGER_INFO.put("NUM_PREC_RADIX", 10);
+        TYPE_INFO.add(SHORT_INFO);
+
+        FLOAT_INFO.put("TYPE_NAME", "Float");
+        FLOAT_INFO.put("DATA_TYPE", Types.REAL);
+        FLOAT_INFO.put("PRECISION", 15);
+        FLOAT_INFO.put("LITERAL_PREFIX", null);
+        FLOAT_INFO.put("LITERAL_SUFFIX", null);
+        FLOAT_INFO.put("CASE_SENSITIVE", false);
+        FLOAT_INFO.put("MINIMUM_SCALE", 0);
+        FLOAT_INFO.put("MAXIMUM_SCALE", 0);
+        FLOAT_INFO.put("NUM_PREC_RADIX", 10);
+        TYPE_INFO.add(FLOAT_INFO);
+
+        DOUBLE_INFO.put("TYPE_NAME", "Double");
+        DOUBLE_INFO.put("DATA_TYPE", Types.DOUBLE);
+        DOUBLE_INFO.put("PRECISION", 15);
+        DOUBLE_INFO.put("LITERAL_PREFIX", null);
+        DOUBLE_INFO.put("LITERAL_SUFFIX", null);
+        DOUBLE_INFO.put("CASE_SENSITIVE", false);
+        DOUBLE_INFO.put("MINIMUM_SCALE", 0);
+        DOUBLE_INFO.put("MAXIMUM_SCALE", 0);
+        DOUBLE_INFO.put("NUM_PREC_RADIX", 10);
+        TYPE_INFO.add(DOUBLE_INFO);
+
+        STRING_INFO.put("TYPE_NAME", "String");
+        STRING_INFO.put("DATA_TYPE", Types.VARCHAR);
+        STRING_INFO.put("PRECISION", Integer.MAX_VALUE);
+        STRING_INFO.put("LITERAL_PREFIX", "'");
+        STRING_INFO.put("LITERAL_SUFFIX", "'");
+        STRING_INFO.put("CASE_SENSITIVE", true);
+        TYPE_INFO.add(STRING_INFO);
+
+        BYTES_INFO.put("TYPE_NAME", "byte[]");
+        BYTES_INFO.put("DATA_TYPE", Types.VARCHAR);
+        BYTES_INFO.put("PRECISION", Integer.MAX_VALUE);
+        BYTES_INFO.put("LITERAL_PREFIX", null);
+        BYTES_INFO.put("LITERAL_SUFFIX", null);
+        BYTES_INFO.put("CASE_SENSITIVE", false);
+        TYPE_INFO.add(BYTES_INFO);
+
+        SQL_DATE_INFO.put("TYPE_NAME", "sql.Date");
+        SQL_DATE_INFO.put("DATA_TYPE", Types.DATE);
+        SQL_DATE_INFO.put("PRECISION", 24);
+        SQL_DATE_INFO.put("LITERAL_PREFIX", null);
+        SQL_DATE_INFO.put("LITERAL_SUFFIX", null);
+        SQL_DATE_INFO.put("CASE_SENSITIVE", false);
+        TYPE_INFO.add(SQL_DATE_INFO);
+
+        UTIL_DATE_INFO.put("TYPE_NAME", "util.Date");
+        UTIL_DATE_INFO.put("DATA_TYPE", Types.DATE);
+        UTIL_DATE_INFO.put("PRECISION", 24);
```

Comment:
What does precision 24 mean for date/time?

Reply:
Beats me - didn't look into it. Probably 24 characters allowed for displaying the date?

This is just pulled from the metadata.

Reply:
Have you looked at what precision means for date/time/timestamp? If not, we should probably correct them for the driver:

The PRECISION column represents the maximum column size that the server supports for the given datatype. For numeric data, this is the maximum precision. For character data, this is the length in characters. For datetime datatypes, this is the length in characters of the String representation (assuming the maximum allowed precision of the fractional seconds component). For binary data, this is the length in bytes. For the ROWID datatype, this is the length in bytes. Null is returned for data types where the column size is not applicable.

Therefore for dates I would expect precision to be 10, time to be 8 + (1 + fractional second precision), timestamp to be 19 + (1 + fractional second precision).

Reply:
Created a backlog ticket